test/e2e/framework/volume/fixtures.go

```diff
@@ -485,9 +485,14 @@ func testVolumeContent(f *framework.Framework, pod *v1.Pod, fsGroup *int64, fsTy
 
 				// Filesystem: check fsType
 				if fsType != "" {
-					ginkgo.By("Checking fsType is correct.")
-					_, err = framework.LookForStringInPodExec(pod.Namespace, pod.Name, []string{"grep", " " + dirName + " ", "/proc/mounts"}, fsType, time.Minute)
-					framework.ExpectNoError(err, "failed: getting the right fsType %s", fsType)
+					// Skip this checking if a Kata Containers pod because the filesystem type
+					// will not match fsType, instead it will be the backend used to share
+					// host filesystems. Currently the guest mounts virtiofs by default.
+					if pod.Spec.RuntimeClassName == nil || !strings.Contains(*pod.Spec.RuntimeClassName, "kata") {
```

Comment:
I don't think we should be encoding hyper specific logic like this inline

I'm not sure we should be doing this at all. Might as well delete the test if we're going to let it sometimes fail.

Either way from a go review standpoint we should keep the default case less indented and return early in the exceptional case.